src/parser/ScriptParser.cpp

```diff
@@ -503,7 +503,9 @@ ScriptParser::InitializeScriptResult ScriptParser::initializeScript(String* sour
 void ScriptParser::generateFunctionByteCode(ExecutionState& state, InterpretedCodeBlock* codeBlock, size_t stackSizeRemain)
 {
 #ifdef ESCARGOT_DEBUGGER
-    ASSERT(m_context->debugger() == nullptr || !m_context->debugger()->enabled());
+    if (m_context->debugger() != nullptr) {
+        m_context->debugger()->setParsingEnabled(false);
```

Comment:
Would you please add comments to elaborate this new code like that `ScriptParser::generateFunctionByteCode` method is invoked only for dynamic functions in debugger mode and debugger ignores these dynamic functions?